specification/workloads/resource-manager/Microsoft.Workloads/connectors/preview/2023-10-01-preview/backupTypes.json

```diff
@@ -0,0 +1,849 @@
+      "required": [
+        "schedulePolicyType"
+      ],
+      "type": "object",
+      "properties": {
+        "schedulePolicyType": {
+          "description": "This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.",
+          "type": "string"
+        }
+      },
+      "discriminator": "schedulePolicyType"
+    },
+    "RetentionPolicy": {
+      "description": "Base class for retention policy.",
+      "required": [
+        "retentionPolicyType"
+      ],
+      "type": "object",
+      "properties": {
+        "retentionPolicyType": {
+          "description": "This property will be used as the discriminator for deciding the specific types in the polymorphic chain of types.",
+          "type": "string"
+        }
+      },
+      "discriminator": "retentionPolicyType"
+    },
+    "AzureVmWorkloadProtectionPolicy": {
+      "description": "Azure VM (Mercury) workload-specific backup policy.",
+      "type": "object",
+      "allOf": [
+        {
+          "$ref": "#/definitions/ProtectionPolicy"
+        }
+      ],
+      "properties": {
+        "workLoadType": {
+          "description": "Type of workload for the backup management",
+          "enum": [
+            "Invalid",
+            "VM",
+            "FileFolder",
+            "AzureSqlDb",
+            "SQLDB",
+            "Exchange",
+            "Sharepoint",
+            "VMwareVM",
+            "SystemState",
+            "Client",
+            "GenericDataSource",
+            "SQLDataBase",
+            "AzureFileShare",
+            "SAPHanaDatabase",
+            "SAPAseDatabase",
+            "SAPHanaDBInstance"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "WorkloadType",
+            "modelAsString": true
+          }
+        },
+        "settings": {
+          "$ref": "#/definitions/Settings",
+          "description": "Common settings for the backup management"
+        },
+        "subProtectionPolicy": {
+          "description": "List of sub-protection policies which includes schedule and retention",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/SubProtectionPolicy"
+          },
+          "x-ms-identifiers": []
+        },
+        "makePolicyConsistent": {
+          "description": "Fix the policy inconsistency",
+          "type": "boolean"
+        }
+      },
+      "x-ms-discriminator-value": "AzureWorkload"
+    },
+    "Settings": {
+      "description": "Common settings field for backup management",
+      "type": "object",
+      "properties": {
+        "timeZone": {
+          "description": "TimeZone optional input as string. For example: TimeZone = \"Pacific Standard Time\".",
+          "type": "string"
+        },
+        "issqlcompression": {
+          "description": "SQL compression flag",
+          "type": "boolean"
+        },
+        "isCompression": {
```

Comment:
>isCompression [](http://example.com/codeflow?start=9&length=13)

A setting named isCompression sounds off. Did you mean this to be an enum Compression with values Enabled/Disabled or a boolean compressionEnabled?

Reply:
Our service is integrated with Azure Backup service. Currently we have defined the backup policy models in our swagger. But before going GA, we would reference the policy models from Azure Backup swagger and not have ownership from our side.
So to avoid renaming of properties in future when we reference these definitions directly from Azure backup service swagger, we have replicated the definitions as how they are in their swagger.

https://github.com/Azure/azure-rest-api-specs/blob/5797d78f04cd8ca773be82d2c99a3294009b3f0a/specification/recoveryservicesbackup/resource-manager/Microsoft.RecoveryServices/stable/2023-04-01/bms.json#L9169